drivers/soundwire/intel.h

```diff
@@ -23,7 +31,7 @@
  * @list: used to walk-through all masters exposed by the same controller
  */
 struct sdw_intel_link_res {
-	struct platform_device *pdev;
+	struct intel_link_dev *ldev;
```

Comment:
I find this odd that struct intel_link_dev has a pointer to struct sdw_intel_link_res which in turn has a pointer to struct intel_link_dev. Is this correct, Pierre?

Reply:
it's the same as in the platform device stuff, but I've now modified the code to avoid such references. I just wanted a first pass that wasn't too intrusive (hence the draft status).